spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -394,14 +393,15 @@ private boolean sendWithObservation(Message<?> message, long timeout) {
 				messageToSendInternal =
 						new ErrorMessage(errorMessage.getPayload(),
 								messageToSend.getHeaders(),
-								errorMessage.getOriginalMessage());
+								Objects.requireNonNull(errorMessage.getOriginalMessage()));
```

Comment:
This is not correct.
That `original` indeed could be null.
We have to look into a ternary operator to chose this or other `ErrorMessage` ctor.